python/paddle/fluid/tests/unittests/test_dist_base.py

```diff
@@ -87,7 +94,7 @@ def run_trainer(self, args):
                                     args.endpoints, args.trainers,
                                     args.sync_mode, args.dc_asgd)
             trainer_prog = t.get_trainer_program()
-        elif args.update_method == "nccl2":
+        elif args.update_method == "nccl2" or args.update_method == "nccl2_reduce_layer":
```

Comment:
maybe `nccl2_reduce_layer` mode should execute the `seq_allreduce_pass` to avoid the NCCL hang ?

Reply:
It will be automatically done if nranks > 1